libs/sdk-core/src/models.rs

```diff
@@ -107,16 +107,56 @@ pub trait SwapperAPI: Send + Sync {
     async fn complete_swap(&self, bolt11: String) -> Result<()>;
 }
 
+/// Details about the BTC/BTC reverse swap pair, at this point in time
+///
+/// Maps the result of https://docs.boltz.exchange/en/latest/api/#getting-pairs for the BTC/BTC pair
 #[derive(Clone, PartialEq, Debug)]
-pub struct ReverseSwapInfo {
+pub struct ReverseSwapPairInfo {
+    /// Minimum amount of sats a reverse swap is allowed to have on this endpoint
     pub min: u64,
+    /// Maximum amount of sats a reverse swap is allowed to have on this endpoint
     pub max: u64,
+    /// Hash of the pair info JSON
     pub fees_hash: String,
+    /// Percentage fee for the Boltz service
     pub fees_percentage: f64,
+    /// Estimated miner fees in sats for locking up funds
     pub fees_lockup: u64,
+    /// Estimated miner fees in sats for claiming funds
     pub fees_claim: u64,
 }
 
+/// Details of past or ongoing reverse swaps, as stored in the Breez local DB
+#[derive(Clone, Debug)]
+pub struct ReverseSwapInfo {
+    // static immutable data
+    pub claim_address: String,
+    pub lockup_address: String,
+
+    // dynamic data
+    pub status: ReverseSwapStatus,
+}
+
+/// The status of a reverse swap
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub enum ReverseSwapStatus {
```

Comment:
@ok300  I believe that as part of monitoring the swap on-chain we will need to update the data such as invoice paid hash, lock_tx_id, claim_tx_id, etc...
So it is early to know if we need these statuses or can derive the state from the data in the db. Ok for now but keep in mind we might (or not) remove these in the future.

Reply:
Good point. Even better if the statuses can be derived.